### PR TITLE
Edit Catalogue solution Implementation

### PR DESCRIPTION
**Pull request by RahulJha-mastek**
Story AB#15297
Task AB#15762

### CODE REVIEW

File: src/NHSD.GPIT.BuyingCatalogue.WebApp/Areas/Admin/Controllers/CatalogueSolutionsController.cs [hunk 1]
```diff
@@ -80,5 +80,63 @@ public async Task<IActionResult> ManageCatalogueSolution(CatalogueItemId solutio
...
+
+        [HttpGet("manage/{solutionId}/description")]
+        public async Task<IActionResult> Description(CatalogueItemId solutionId)
+        {
+            var solution = await solutionsService.GetSolution(solutionId);
+
+            if (solution == null)
+                return BadRequest($"No Solution found for Id: {solutionId}");
+
+            var solutionDescription = new DescriptionModel(solution);
```
Comment (PhysicsJon) [inline src/NHSD.GPIT.BuyingCatalogue.WebApp/Areas/Admin/Controllers/CatalogueSolutionsController.cs:92]: Do we need the assignment to a variable before immediately returning it to the view?

File: src/NHSD.GPIT.BuyingCatalogue.WebApp/Areas/Admin/Controllers/CatalogueSolutionsController.cs [hunk 2]
```diff
@@ -80,5 +80,63 @@ public async Task<IActionResult> ManageCatalogueSolution(CatalogueItemId solutio
...
+
+        [HttpGet("manage/{solutionId}/implementation")]
+        public async Task<IActionResult> Implementation(CatalogueItemId solutionId)
+        {
+            var solution = await solutionsService.GetSolution(solutionId);
+
+            if (solution == null)
+                return BadRequest($"No Solution found for Id: {solutionId}");
+
+            var implementationTimescale = new ImplementationTimescaleModel(solution);
```
Comment (PhysicsJon) [inline src/NHSD.GPIT.BuyingCatalogue.WebApp/Areas/Admin/Controllers/CatalogueSolutionsController.cs:122]: Do we need the assignment to a variable before immediately returning it to the view?

File: tests/NHSD.GPIT.BuyingCatalogue.WebApp.UnitTests/Areas/Admin/Controllers/CatalogueSolutionsControllerTests.cs [hunk 3]
```diff
@@ -206,5 +206,210 @@ public static void Constructor_NullService_ThrowsException()
             model.Solution.Should().BeEquivalentTo(expected);
             model.LastUpdatedByName.Should().BeEquivalentTo($"{aspNetUser.FirstName} {aspNetUser.LastName}");
         }
+
+        [Fact]
+        public static async Task Get_Description_ReturnsViewWithExpectedModel()
+        {
+            var expected = new CatalogueItem
+            {
+                CatalogueItemId = new EntityFramework.Ordering.Models.CatalogueItemId(999999, "999"),
```
Comment (PhysicsJon) [inline tests/NHSD.GPIT.BuyingCatalogue.WebApp.UnitTests/Areas/Admin/Controllers/CatalogueSolutionsControllerTests.cs:215]: Why is the CatalogueItemId being fully qualified here, yet not lower down in the class?

File: tests/NHSD.GPIT.BuyingCatalogue.WebApp.UnitTests/Areas/Admin/Controllers/CatalogueSolutionsControllerTests.cs [hunk 4]
```diff
@@ -206,5 +206,210 @@ public static void Constructor_NullService_ThrowsException()
...
+                    Summary = "XYZ Summary",
+                    FullDescription = "XYZ description",
+                    AboutUrl = "Fake url",
+                },
+                Name = "Fake Solution",
+            };
+
+            var mockSolutionService = new Mock<ISolutionsService>();
+            var mockUserService = new Mock<IUsersService>();
+            mockSolutionService.Setup(s => s.GetSolution(It.IsAny<EntityFramework.Ordering.Models.CatalogueItemId>()))
```
Comment (PhysicsJon) [inline tests/NHSD.GPIT.BuyingCatalogue.WebApp.UnitTests/Areas/Admin/Controllers/CatalogueSolutionsControllerTests.cs:227]: Why is the CatalogueItemId being fully qualified here, yet not lower down in the class?

**Review by PhysicsJon — CHANGES_REQUESTED** [review]

File: src/NHSD.GPIT.BuyingCatalogue.WebApp/Areas/Admin/Controllers/CatalogueSolutionsController.cs (hunk 1 above)
Comment (RahulJha-mastek) [reply src/NHSD.GPIT.BuyingCatalogue.WebApp/Areas/Admin/Controllers/CatalogueSolutionsController.cs]: Updated

File: src/NHSD.GPIT.BuyingCatalogue.WebApp/Areas/Admin/Controllers/CatalogueSolutionsController.cs (hunk 2 above)
Comment (RahulJha-mastek) [reply src/NHSD.GPIT.BuyingCatalogue.WebApp/Areas/Admin/Controllers/CatalogueSolutionsController.cs]: Updated

File: tests/NHSD.GPIT.BuyingCatalogue.WebApp.UnitTests/Areas/Admin/Controllers/CatalogueSolutionsControllerTests.cs (hunk 3 above)
Comment (RahulJha-mastek) [reply tests/NHSD.GPIT.BuyingCatalogue.WebApp.UnitTests/Areas/Admin/Controllers/CatalogueSolutionsControllerTests.cs]: Removed fully qualified name

File: tests/NHSD.GPIT.BuyingCatalogue.WebApp.UnitTests/Areas/Admin/Controllers/CatalogueSolutionsControllerTests.cs (hunk 4 above)
Comment (RahulJha-mastek) [reply tests/NHSD.GPIT.BuyingCatalogue.WebApp.UnitTests/Areas/Admin/Controllers/CatalogueSolutionsControllerTests.cs]: Removed fully qualified name

**Review by PhysicsJon — APPROVED** [review]

**Pull request merged**